packages/liquidation-sdk-viem/src/preLiquidation/logGetters.ts

```diff
@@ -0,0 +1,207 @@
+import type { Account, Chain, Client, Transport } from "viem";
+import { getBlockNumber, getLogs } from "viem/actions";
+import { preLiquidationFactoryConfigs } from "../addresses";
+import type { PreLiquidation } from "./types";
+
+export async function preLiquidationLogs<chain extends Chain = Chain>(
+  client: Client<Transport, chain, Account>,
+): Promise<PreLiquidation[]> {
+  const chainId = ChainUtils.parseSupportedChainId(client.chain.id);
+
+  try {
+    const head = await getBlockNumber(client);
+    const intervals = sliceBlockInterval(
+      preLiquidationFactoryConfigs[chainId].startBlock,
+      BigInt(head),
+    );
+
+    const logs = (
+      await Promise.all(
+        intervals.map((interval) =>
+          getLogs(client, {
+            address: preLiquidationFactoryConfigs[chainId].address,
+            fromBlock: interval.startBlock,
+            toBlock: interval.endBlock,
+            event: {
+              type: "event",
+              name: "CreatePreLiquidation",
+              inputs: [
+                {
+                  name: "preLiquidation",
+                  type: "address",
+                  indexed: true,
+                  internalType: "address",
+                },
+                {
+                  name: "id",
+                  type: "bytes32",
+                  indexed: false,
+                  internalType: "Id",
+                },
+                {
+                  name: "preLiquidationParams",
+                  type: "tuple",
+                  indexed: false,
+                  internalType: "struct PreLiquidationParams",
+                  components: [
+                    {
+                      name: "preLltv",
+                      type: "uint256",
+                      internalType: "uint256",
+                    },
+                    {
+                      name: "preLCF1",
+                      type: "uint256",
+                      internalType: "uint256",
+                    },
+                    {
+                      name: "preLCF2",
+                      type: "uint256",
+                      internalType: "uint256",
+                    },
+                    {
+                      name: "preLIF1",
+                      type: "uint256",
+                      internalType: "uint256",
+                    },
+                    {
+                      name: "preLIF2",
+                      type: "uint256",
+                      internalType: "uint256",
+                    },
+                    {
+                      name: "preLiquidationOracle",
+                      type: "address",
+                      internalType: "address",
+                    },
+                  ],
+                },
+              ],
+            },
+          }),
+        ),
+      )
+    ).flat();
+
+    return logs.map((log) => {
+      return {
+        marketId: log.args.id! as MarketId,
+        address: log.args.preLiquidation! as Address,
```

Comment:
I believe there's a `strict: true` field in the viem request that will make these non-optional automagically.